integration-tests/regression.test.js

```diff
@@ -163,4 +163,59 @@ describe('Tram-One - regressions', () => {
 		// cleanup - remove app
 		appContainer.remove()
 	})
+
+	it('should keep focus when both the parent and child element would update', async () => {
+		// for focus to work correctly, the element needs to be attached to the document
+		const appContainer = document.createElement('div')
+		appContainer.id = 'app'
+		window.document.body.appendChild(appContainer)
+
+		// start the app using a css selector
+		startApp('#app')
+
+		// previously when interacting with an input, if both a parent and child element
+		// would update, then focus would not reattach
+
+		// focus on the parent input
+		userEvent.click(getByLabelText(appContainer, 'Mirror Input'))
+
+		// verify that the element has focus (before we start changing text)
+		await waitFor(() => {
+			expect(getByLabelText(appContainer, 'Mirror Input')).toHaveFocus()
+		})
+
+		// update the state by typing
+		userEvent.type(getByLabelText(appContainer, 'Mirror Input'), 'Test')
+
+		// verify the element has the new value
+		expect(getByLabelText(appContainer, 'Mirror Input')).toHaveValue('Test')
+
+		// wait for the sub element to also have the same value
+		await waitFor(() => {
+			expect(getByLabelText(appContainer, 'Sub Mirror Input')).toHaveValue('Test')
+		})
+
+		// repeat the test with the child element
+		// focus on the child input
+		userEvent.click(getByLabelText(appContainer, 'Sub Mirror Input'))
+
+		// verify that the element has focus (before we start changing text)
+		await waitFor(() => {
+			expect(getByLabelText(appContainer, 'Sub Mirror Input')).toHaveFocus()
+		})
+
+		// update the state by typing
+		userEvent.type(getByLabelText(appContainer, 'Sub Mirror Input'), ' Again')
+
+		// verify the element has the new value
+		expect(getByLabelText(appContainer, 'Sub Mirror Input')).toHaveValue('Test Again')
```

Comment:
I can't help but feel like this test gets into a bit of the internals of the side effects, and less about what is critical for the test. I think it's fine, but years down the line will this part of the test make sense?

Reply:
`toHaveValue` is totally fine, that is expected for an input to work.
Nothing here is working around the internals, it's all just labels and inputs (as you might interact with them in react)

Reply:
I meant more so that all this test should be checking for is that focus is retained. By having more code it could make it annoying to see where it failed at a glance. But I see what you're saying.

Reply:
oh, yeah, I guess, I was seeing two issues, but I can update the test to also verify that the focus stays (I do that in the test before as well, so I have it already setup).